base/cbgt.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2023-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package base
+
+import (
+	"fmt"
+
+	"github.com/couchbase/cbgt"
+)
+
+// NewCbgtCfgMem runs cbgt.NewCfgMem and sets the matching version number we expect for Sync Gateway.
+func NewCbgtCfgMem() (*cbgt.CfgMem, error) {
+	cfg := cbgt.NewCfgMem()
+	cas, err := cfg.Set(cbgt.VERSION_KEY, []byte(SGCbgtMetadataVersion), 0)
```

Comment:
I would expect cbgt to be writing the version key, in the same way it does when running against a document-backed store.  Why do we need to define it in advance in the Cfg key value store when using CfgMem?

Reply:
The spec of the implmentation is such that it reads the key out of the config if it's not set, so this matches our not in memory version https://github.com/couchbase/sync_gateway/blob/master/base/sg_cluster_cfg.go#L69

Reply:
Makes sense - thanks.